org.eclipse.dawnsci.hdf.object.test/src/org/eclipse/dawnsci/hdf/object/AbstractThreadTestBase.java

```diff
@@ -0,0 +1,90 @@
+/*-
+ *******************************************************************************
+ * Copyright (c) 2011, 2017 Diamond Light Source Ltd.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Matthew Gerring - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+
+package org.eclipse.dawnsci.hdf.object;
+
+/**
+ * Class to extend, does not have any tests itself.
+ */
+public abstract class AbstractThreadTestBase {
+
+	public void testInTestThread() throws Throwable {
+		doTestOfDataSet(0);
+
+		System.out.println("One thread passed.");
+	}
+
+	protected abstract void doTestOfDataSet(int index) throws Throwable;
+
+	private Throwable exception;
+
+	/**
+	 * Test shows that nexus API cannot handle threads.
+	 * 
+	 * @throws Exception
+	 */
+	public void testWithTenThreads() throws Throwable {
+
+		testWithNThreads(10);
+	}
+
+	/**
+	 * Start threads that each run {@link #doTestOfDataSet(int)}
+	 * 
+	 * @param threadNumber
+	 * @throws Throwable
+	 */
+	protected void testWithNThreads(int threadNumber) throws Throwable {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)